docs/tutorials/wiki2/authentication.rst

```diff
@@ -215,6 +197,9 @@ This code adds three new views to the application:
   The check is done by first finding a ``User`` record in the database, then
   using our ``user.check_password`` method to compare the hashed passwords.
 
+  At a privilege boundary we are sure to reset the CSRF token using :meth:`pyramid.csrf.new_csrf_token`.
```

Comment:
Could you add a line to explain why this is useful?

Reply:
I would prefer to just provide the pattern. I think it's distracting to go into too many details in the tutorial. I'd accept something if you added one, especially to the CSRF protection docs in the security chapter - but ideally not here.

Reply:
I couldn’t add something, as I don’t understand the reason now :slightly_smiling_face: 

Thanks for showing the pattern nonetheless!  I don’t see harm in copying it without full understanding, and I may understand the reasons after I read some articles in my reading list.

Reply:
It's definitely not as important for a CSRF token - which mostly needs to be unique per client so that an attacking website cannot guess it. It's way more important for session data. The general recommendation is to always clear application state when someone logs in or logs out. You don't want to accidentally access random state with a logged-in user that was set by someone else before the user logged in. It can trigger unintended side-effects and unanticipated behaviors.

Reply:
Thanks for explaining.  I haven’t encountered these issues before, as I work on apps that don’t use sessions (clients drive interactions, or if state has to be persistent then it’s application state that is saved in database).  I guess I haven’t had any fun use case yet like «items added to cart before login should be kept after»!